core/capabilities/ccip/configs/solana/contract_reader.go

```diff
@@ -1,11 +1,488 @@
+		Value     *big.Int `json:"value"`
+		Timestamp uint32   `json:"timestamp"`
+	}
+
+	var offRampIDL solanacodec.IDL
+	if err := json.Unmarshal([]byte(ccipOffRampIDL), &offRampIDL); err != nil {
+		return config.ContractReader{}, fmt.Errorf("unexpected error: invalid CCIP OffRamp IDL, error: %w", err)
+	}
+
+	var feeQuoterIDL solanacodec.IDL
+	// TODO ccipFeeQuoterIDL needs to be exported from chainlink-ccip, this is just a placeholder
+	if err := json.Unmarshal([]byte(ccipFeeQuoterIDL), &feeQuoterIDL); err != nil {
+		return config.ContractReader{}, fmt.Errorf("unexpected error: invalid CCIP Fee Quoter IDL, error: %w", err)
+	}
+
+	var routerIDL solanacodec.IDL
+	if err := json.Unmarshal([]byte(ccipRouterIDL), &routerIDL); err != nil {
+		return config.ContractReader{}, fmt.Errorf("unexpected error: invalid CCIP Router IDL, error: %w", err)
+	}
+
+	locationFirst := codec.ElementExtractorLocationFirst
+	return config.ContractReader{
+		AddressShareGroups: [][]string{{consts.ContractNameRouter, consts.ContractNameNonceManager}},
+		Namespaces: map[string]config.ChainContractReader{
+			consts.ContractNameOffRamp: {
+				IDL: offRampIDL,
+				Reads: map[string]config.ReadDefinition{
+					consts.MethodNameGetLatestPriceSequenceNumber: {
+						ChainSpecificName: "GlobalState",
+						ReadType:          config.Account,
+						PDADefinition:     solanacodec.PDATypeDef{Prefix: []byte("state")},
+						OutputModifications: codec.ModifiersConfig{
+							&codec.RenameModifierConfig{
+								Fields: map[string]string{"LatestPriceSequenceNumber": "LatestSeqNr"},
+							}},
+					},
+					consts.MethodNameOffRampGetStaticConfig: {
+						ChainSpecificName: "Config",
+						ReadType:          config.Account,
+						PDADefinition: solanacodec.PDATypeDef{
+							Prefix: []byte("config"),
+						},
+						OutputModifications: codec.ModifiersConfig{
+							&codec.RenameModifierConfig{
+								Fields: map[string]string{
+									"SvmChainSelector": "ChainSelector",
+								},
+							},
+							&codec.HardCodeModifierConfig{
+								OnChainValues: map[string]any{
+									"GasForCallExactCheck": 0,
+									"RmnRemote":            []byte{},
+									// TODO what to do with these two? Do they share address with router?
+									"TokenAdminRegistry": []byte{},
+									"NonceManager":       []byte{},
+								},
+							},
+						},
+					},
+					consts.MethodNameOffRampGetDynamicConfig: {
+						ChainSpecificName: "Config",
+						ReadType:          config.Account,
+						PDADefinition: solanacodec.PDATypeDef{
+							Prefix: []byte("config"),
+						},
+						OutputModifications: codec.ModifiersConfig{
+							&codec.RenameModifierConfig{
+								Fields: map[string]string{"EnableManualExecutionAfter": "PermissionLessExecutionThresholdSeconds"},
+							},
+							&codec.HardCodeModifierConfig{
+								OnChainValues: map[string]any{
+									"IsRMNVerificationDisabled": false,
+									// TODO what to do with this address? Is it same as Router?
```

Comment:
"Optional message interceptor to validate messages. Zero address = no interceptor." 

I'm not sure this is available on Solana, set it to `0x0`?